### PR TITLE
ENH: devel upload dandiset metadata

### LINKED ISSUE #339: upload --upload-dandiset-yaml: to ease/automate migration - allow for upload of (migrated) dandiset.yaml
Assuming that dandi-api instance has all dandisets "pre-generated", we could do  
```
dandi download https://gui.dandiarchive.org/#/dandiset/$ds
cd $ds
DANDI_DEVEL=1 dandi upload -i dandi-api --upload-dandiset-yaml
```
and it would "do the right thing" for migration - load old style dandiset.yaml, convert, update metadata for the dandiset (ATM we do not upload `dandiset.yaml`)  (in reality we will just run `dandi upload` on already fetched datalad datasets, so no download would really be necessary

### PR DESCRIPTION
**Pull request by yarikoptic**
Closes #339 
ref https://github.com/dandi/dandi-api/issues/66 on not forgetting the trailing `/`

TODO
- [x] test

### CODE REVIEW

File: dandi/dandiapi.py [hunk 1]
```diff
@@ -123,7 +122,9 @@ def send_request(
         url = self.get_url(path)
 
         # Make the request, passing parameters and authentication info
-        _headers = headers or {}
+        _headers = dict(self._headers)
+        if headers:
+            _headers.update(_headers)
```
Comment (jwodder) [inline dandi/dandiapi.py:127]: The second `_headers` should not have a leading underscore.

File: dandi/upload.py [hunk 2]
```diff
@@ -56,9 +57,13 @@ def upload(
             validation,
             dandiset_path,
             allow_any_path,
+            upload_dandiset_metadata,
             devel_debug,
         )
 
+    if upload_dandiset_metadata:
+        raise NotImplementedError("Not supported against original Girder based server")
```
Comment (jwodder) [inline dandi/upload.py:65]: The error message should say what exactly it is that's not supported.

**Review by jwodder — CHANGES_REQUESTED** [review]

File: dandi/dandiapi.py (hunk 1 above)
Comment (yarikoptic) [reply dandi/dandiapi.py:127]: d'oh, right -- thanks! will fix

File: dandi/upload.py (hunk 2 above)
Comment (yarikoptic) [reply dandi/upload.py:65]: good point, thanks

**Comment by yarikoptic** [comment]
force pushed updates

**Comment by jwodder** [comment]
@yarikoptic How do I mark this PR as "approved"?  Or should I just go ahead and merge it?

**Comment by yarikoptic** [comment]
> @yarikoptic How do I mark this PR as "approved"? 

Click on green "Review changes"  in "Files changed" tab, choose "Approve"

> Or should I just go ahead and merge it?

In general/future -- you are most welcome to do just that if satisfied, unless multiple reviewers were requested and we are still waiting on feedback.  Meanwhile, I will beat you to it, and will click Merge ;)

**Pull request merged**